### PR TITLE
(#3452) added an option to append --build / --no-build to InspectCodeSettings

### LINKED ISSUE #3452: Missing option in InspectCodeSettings: `--build` and `--no-build` flags
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool, Cake Frosting, Cake runner for .NET Framework, Cake runner for .NET Core

### Cake version

1.2.0

### Operating system

N/A

### Operating system architecture

N/A

### CI Server

_No response_

### What are you seeing?

Running `InspectCode` with `JetBrains.ReSharper.CommandLineTools` version 2021.2.0 or later produces the following warning:

```
Warning: Starting from version 2021.2, InspectCode builds the target solution before starting the analysis to make sure it only finds relevant code issues.
To explicitly accept the new behavior and suppress this warning, use the '--build' option.
To match the behavior in previous versions and skip the build, use '--no-build'.
```

However, currently `InspectCodeSettings` does not have a property to pass those flags.

### What is expected?

I propose adding a new property to `InspectCodeSettings`: `BuildBehaviour` which is an enum containing of three values: `Undefined`, `Build` and `NoBuild`.

The default should be `Undefined` which will result in no change to the generated command arguments and hence be backwards compatible with existing usages.
Setting `BuildBehaviour` to `Build` should add the `--build` flag to the generated command arguments, and setting  `BuildBehaviour` to `NoBuild` should add the `--no-build` flag to the generated command arguments.

### Steps to Reproduce

Select any build that uses `InspectCode` and bump  `JetBrains.ReSharper.CommandLineTools` to version 2021.2.0 or later.

### Output log

_No response_

### PR DESCRIPTION
**Pull request by nils-a**
fixes #3452

### CODE REVIEW

File: src/Cake.Common.Tests/Unit/Tools/InspectCode/InspectCodeRunnerTests.cs [hunk 1]
```diff
@@ -408,6 +408,48 @@ public void Should_Use_Provided_Config_File()
                 // Then
                 Assert.Equal("\"/config=/Working/config.xml\"", result.Args);
             }
+
+            [Fact]
+            public void Should_Contain_build_or_noBuild_if_Build_is_not_set()
```
Comment (augustoproiete) [inline src/Cake.Common.Tests/Unit/Tools/InspectCode/InspectCodeRunnerTests.cs:413]:
```suggestion
            public void Should_Not_Contain_build_nor_noBuild_if_Build_is_not_set()
```

**Review by augustoproiete — CHANGES_REQUESTED** [review]
LGTM. Small nitpick on the of one of the tests to reflect what it's testing

File: src/Cake.Common.Tests/Unit/Tools/InspectCode/InspectCodeRunnerTests.cs (hunk 1 above)
Comment (nils-a) [reply src/Cake.Common.Tests/Unit/Tools/InspectCode/InspectCodeRunnerTests.cs]: Good catch. Fixed.

**Review by augustoproiete — APPROVED** [review]
LGTM

**Pull request merged**